CMakeLists.txt

```
@@ -76,30 +79,38 @@ endif ( )
 #
 file( GLOB_RECURSE MANIFEST "${SOURCE_DIR}/*.cpp" )
 
-set( STATIC_LIBRARY_NAME "${PROJECT_NAME}-static" )
-add_library( ${STATIC_LIBRARY_NAME} STATIC ${MANIFEST} )
-set_property( TARGET ${STATIC_LIBRARY_NAME} PROPERTY CXX_STANDARD 14 )
-set_property( TARGET ${STATIC_LIBRARY_NAME} PROPERTY CXX_STANDARD_REQUIRED ON )
-set_target_properties( ${STATIC_LIBRARY_NAME} PROPERTIES OUTPUT_NAME ${PROJECT_NAME} )
-	
-set( SHARED_LIBRARY_NAME "${PROJECT_NAME}-shared" )
-add_library( ${SHARED_LIBRARY_NAME} SHARED ${MANIFEST} )
-set_property( TARGET ${SHARED_LIBRARY_NAME} PROPERTY CXX_STANDARD 14 )
-set_property( TARGET ${SHARED_LIBRARY_NAME} PROPERTY CXX_STANDARD_REQUIRED ON )
-if ( WIN32 )
-	# Workaround to avoid name clash of static lib and dynamic import lib under windows.
-	set_target_properties( ${SHARED_LIBRARY_NAME} PROPERTIES OUTPUT_NAME "${PROJECT_NAME}-shared" )
-else ( )
-	set_target_properties( ${SHARED_LIBRARY_NAME} PROPERTIES OUTPUT_NAME ${PROJECT_NAME} )
-endif ( )	
-set_target_properties( ${SHARED_LIBRARY_NAME} PROPERTIES SOVERSION ${PROJECT_VERSION_MAJOR} VERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR} )
+if ( BUILD_STATIC_LIBRARY )
+    set( STATIC_LIBRARY_NAME "${PROJECT_NAME}-static" )
+    add_library( ${STATIC_LIBRARY_NAME} STATIC ${MANIFEST} )
+    set_property( TARGET ${STATIC_LIBRARY_NAME} PROPERTY CXX_STANDARD 14 )
+    set_property( TARGET ${STATIC_LIBRARY_NAME} PROPERTY CXX_STANDARD_REQUIRED ON )
+    set_target_properties( ${STATIC_LIBRARY_NAME} PROPERTIES OUTPUT_NAME ${PROJECT_NAME} )
+
+    if ( BUILD_SSL )
+        target_link_libraries( ${STATIC_LIBRARY_NAME} LINK_PRIVATE ${ssl_LIBRARY_STATIC} ${crypto_LIBRARY_STATIC} ${CMAKE_DL_LIBS} )
+    else ( )
+        target_link_libraries( ${STATIC_LIBRARY_NAME} ${CMAKE_DL_LIBS} )
+    endif ( )
+endif ( )
 
-if ( BUILD_SSL )
-    target_link_libraries( ${SHARED_LIBRARY_NAME} LINK_PRIVATE ${ssl_LIBRARY_SHARED} ${crypto_LIBRARY_SHARED} )
-    target_link_libraries( ${STATIC_LIBRARY_NAME} LINK_PRIVATE ${ssl_LIBRARY_STATIC} ${crypto_LIBRARY_STATIC} ${CMAKE_DL_LIBS} )
-else ( )
-    target_link_libraries( ${SHARED_LIBRARY_NAME} )
-    target_link_libraries( ${STATIC_LIBRARY_NAME} ${CMAKE_DL_LIBS} )
+if ( BUILD_SHARED_LIBRARY )
+    set( SHARED_LIBRARY_NAME "${PROJECT_NAME}-shared" )
+    add_library( ${SHARED_LIBRARY_NAME} SHARED ${MANIFEST} )
+    set_property( TARGET ${SHARED_LIBRARY_NAME} PROPERTY CXX_STANDARD 14 )
+    set_property( TARGET ${SHARED_LIBRARY_NAME} PROPERTY CXX_STANDARD_REQUIRED ON )
+    if ( WIN32 )
+        # Workaround to avoid name clash of static lib and dynamic import lib under windows.
+        set_target_properties( ${SHARED_LIBRARY_NAME} PROPERTIES OUTPUT_NAME "${PROJECT_NAME}-shared" )
```

Only MSVC uses file names that conflict between shared and static libraries.
Also, the `.dll` does not really need the `-shared` suffix.
So set the `ARCHIVE_OUTPUT_NAME` property additionally.
```suggestion
    if ( MSVC )
        # Workaround to avoid name clash of static lib and dynamic import lib under windows.
        set_target_properties( ${SHARED_LIBRARY_NAME} PROPERTIES OUTPUT_NAME ${PROJECT_NAME} ARCHIVE_OUTPUT_NAME "${PROJECT_NAME}-shared" )
```

This results in the following files:
```
restbed-shared.lib # import library of restbed.dll (shared library)
restbed.dll        # shared library
restbed.lib        # static library
```

---

Thanks for your feedback. But I'm not sure what the proposed change of yours has to do with my Pull Request. I just have introduced three boolean switches (which default to ON) and have surrounded already existing CMake code with if ( <switch> ) ... endif() (and have indented the lines within it), that's all. It's a very trivial change which should have no impact except offering the possibility to set one (or more) of these new switches to OFF. Therefore I have no idea why one of the tests fails when run on MS-Windows.

Regarding the MS-Windows build: I can neither comment on this nor check it since I'm using Linux only and have no experience with CMake and MS-Windows. But I don't see the correlation with my changes, and furthermore it's a test execution that fails under MS-Windows, not the build, at least that is my understanding.